docs/architecture/adr-031-msg-service.md

```diff
@@ -8,8 +8,14 @@
 
 Proposed
 
+## Abstract
+
+We want to reuse protobuf code generator to crate for use _factories_ for `Msg` and message handler `Responses`. Specifically we can use a protobuf `service` which will provide us library functions to be used internally an in external libraries.
+
 ## Context
 
+TODO: Explain what's wrong with creating a message.
```

Comment:
I'm not sure quite what you mean by this @robert-zaremba ? I don't think there's actually anything wrong with creating messages, this just improves developer UX.

Reply:
I did add a bit more to context anyway though. Maybe take another look?

Reply:
Looks good. Thanks.